core-api/src/main/kotlin/com/jakemoore/datakache/core/connections/mongo/changestream/ChangeStreamStateManager.kt

```
@@ -0,0 +1,165 @@
+    /**
+     * Cancels and clears job references.
+     */
+    suspend fun cancelJobs() {
+        changeStreamJob?.cancel()
+        eventProcessorJob?.cancel()
+
+        // Wait for jobs to complete
+        try {
+            changeStreamJob?.join()
+            eventProcessorJob?.join()
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)